pkg/api/context.go

```diff
@@ -26,6 +27,9 @@ import (
 // Context carries values across API boundaries.
 type Context interface {
 	Value(key interface{}) interface{}
+	Deadline() (deadline time.Time, ok bool)
```

Comment:
These three fields don't seem to be used-- can we save them until the PR where they start getting used (so we can talk about the use case before adding).